pkg/providers/instancetype/instancetype.go

```diff
@@ -247,6 +261,52 @@ func (p *DefaultProvider) UpdateInstanceTypeOfferings(ctx context.Context) error
 	return nil
 }
 
+func (p *DefaultProvider) UpdateVMCapacityCache(ctx context.Context, kubeClient client.Client, nodeName string) error {
+
+	node := &corev1.Node{}
+	if err := kubeClient.Get(ctx, client.ObjectKey{Name: nodeName}, node); err != nil {
+		return fmt.Errorf("failed to get node: %w", err)
+	}
+
+	nodeClaimList := &karpv1.NodeClaimList{}
+	if err := kubeClient.List(ctx, nodeClaimList); err != nil {
+		return fmt.Errorf("failed to list nodeclaims: %w", err)
+	}
+	nodeClaim, found := lo.Find(nodeClaimList.Items, func(nc karpv1.NodeClaim) bool {
+		return nc.Status.NodeName == node.Name
+	})
+	if !found {
+		return fmt.Errorf("failed to find nodeclaim")
+	}
+
+	nodeClass := &v1.EC2NodeClass{}
+	if err := kubeClient.Get(ctx, client.ObjectKey{Name: nodeClaim.Spec.NodeClassRef.Name}, nodeClass); err != nil {
+		return fmt.Errorf("failed to get ec2nodeclass: %w", err)
+	}
+
+	// Ensure AMI is current
+	if !lo.ContainsBy(nodeClass.Status.AMIs, func(ami v1.AMI) bool {
+		return ami.ID == nodeClaim.Status.ImageID
+	}) {
+		return nil
+	}
+
+	instanceType := node.Labels[corev1.LabelInstanceTypeStable]
+
+	actualCapacity := node.Status.Capacity.Memory()
```

Comment:
This could later be expanded to caching a v1.ResourceList but should probably be limited to those that are known to be safe overrides.